src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -260,10 +260,14 @@ public Response execute(Command command) throws WebDriverException {
 
         if (DriverCommand.NEW_SESSION.equals(command.getName())
                 && getCommandCodec() instanceof W3CHttpCommandCodec) {
-            setCommandCodec(new AppiumW3CHttpCommandCodec());
-            getAdditionalCommands().forEach(this::defineCommand);
+            configureW3CMode();
         }
 
         return response;
     }
+    
+    public void configureW3CMode() {
```

Comment:
configureW3CMode -> I would rather change it to `configureCooandCodec` and pass the codec instance as an argument

Reply:
Sure. Makes sense.